velox/functions/sparksql/String.h

```diff
@@ -1130,4 +1130,52 @@ struct ReplaceFunction {
   }
 };
 
+template <typename T>
+struct FindInSetFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<int32_t>& result,
+      const arg_type<Varchar>& str,
+      const arg_type<Varchar>& strArray) {
+    int32_t index = 1;
+    int32_t lastComma = -1;
+
+    auto match = std::string_view(str);
+    auto arrayStr = std::string_view(strArray);
```

Comment:
I think we can directly use `strArray` to call `compare()`. It's `StringView`. If so, we can remove this line.

Reply:
The StringView compare method takes only one parameter for comparison, which is not what I want.

Reply:
> The StringView compare method takes only one parameter for comparison, which is not what I want.

I need to compare characters of the specified length in `strArray`, such as `find_in_set("d","abc,d,e,f")`, I need to pick up `abc` to copare with `d`, and then pick `d` to compare with `str`, and cycle like this. So The StringView's compare method is doesn't meet my needs.

Reply:
One way is to construct a StringView for a substring, e.g, `StringView(strArray.data() + offset, len)`, then we can directly use "==" operator to compare it with `str`. `StringView`'s compare() is also good, but looks "==" is more straightforward.
https://github.com/facebookincubator/velox/blob/main/velox/type/StringView.h#L134

Reply:
> One way is to construct a StringView for a substring, e.g, `StringView(strArray.data() + offset, len)`, then we can directly use "==" operator to compare it with `str`. `StringView`'s compare() is also good, but looks "==" is more straightforward. 

Thanks, I think this  will generates a lot of substrings, it's depend on the number of `,`,  If so, it's not the best way. What do you think？

Reply:
> Thanks, I think this will generates a lot of substrings, it's depend on the number of `,`, If so, it's not the best way. What do you think？
It seems that we can avoid this issue, let me implement it first.Thank you for review.

Reply:
It will not copy the substring, but only creates some extra StringView instances. 
I note StringView is velox-optimized for better comparison. I'm not sure how much performance gain it will bring out. Maybe, you can just use any one (StringView or std::string_view for comparing) and let masha decide in final review. Thanks!

Reply:
Ok, thanks. I'm going to try to optimize with StringView.

Reply:
I've already used StringView, I need to test it, wait CI success . Thanks

Reply:
All CI tests passed.